lib/tasks/remove_invalid_cases.rake

```diff
@@ -0,0 +1,157 @@
+require 'csv'
+require 'json'
+
+namespace :cases do
+
+  namespace :validation do 
+
+    desc "Validate whether the case in database meets the validation of Case Model."
+    task :validate, [:file] => :environment do |_task, args|
+      raise "Must specify the csv file for outputing the result " if args[:file].blank?
+      counter = 0
+
+      CSV.open(args[:file], "wb") do |csv|
+        csv << ["case_number", "errors"]
+        counter = 0
+        records_array = cases_scope
+        records_array.each do | record |
+          puts "Checking case #{record["number"]}"
+          errors = validate_case(record)
+          if errors.present?
+            counter += 1
+            csv << [record["number"], errors]
+          end
+        end
+      end 
+      puts "Totally #{counter} cases couldn't pass the validation."
+    end
+
+    desc "Remove invalid case based on the validation of Case Model."
+    task :remove => :environment do
+      if is_on_production?
+        puts "Cannot run this command on production environment!"
+      else
+        question_user("Are you sure this is not production env and want to remove those invalid cases (y/n)?")
+        counter = 0
+        records_array = cases_scope
+        records_array.each do | record |
+          puts "Checking case #{record["number"]}"
+          case_id = record['id']
+          errors = validate_case(record)
+          if errors.present?
+            counter += 1
+            remove_case(case_id)
+            puts "Removed case #{record["number"]}"
+          end
+        end
+      end
+      puts "Totally #{counter} invalid cases have been removed."
+    end
+
+    private 
+
+    def is_on_production?
+      ENV['ENV'].present? && ENV['ENV'] == 'prod'
```

Comment:
Doesn't everything on the test environments run in 'prod' mode? Would it not be better to flag based on the host instead like in `Settings.yml` with the feature flags 🤔

Reply:
I think you may get it mixed with another environment variable RAILS_ENV,   which has 2 values 'development' and 'production' and they are for rails's own running mode.   but  ENV['ENV'] is set from our config.yaml which is about which cloud environment this app is in and has values of 'staginig', 'production' and 'development' etc.

Reply:
Yep thats it 🤦